saleor/discount/utils/order.py

```diff
@@ -251,3 +315,49 @@ def create_or_update_line_discount_objects_for_manual_discounts(lines_info):
 
     if discount_to_update:
         OrderLineDiscount.objects.bulk_update(discount_to_update, ["amount_value"])
+
+
+def create_order_line_discount_objects_for_catalogue_promotions(
+    line: "OrderLine",
+    rules_info: Iterable[VariantPromotionRuleInfo],
+    channel: Channel,
+) -> Iterable["OrderLineDiscount"]:
+    from ...order.utils import order_qs_select_for_update
+
+    line_discounts_to_create_inputs: list[dict] = []
+    for rule_info in rules_info:
```

Comment:
As there is assumption that only one catalogue discount object can be created, why are you iterating here through all `rules_info`? Maybe line field should be changed from `rules_info` to `rule_info` - single one, instead of list.

Reply:
Our whole codebase assumes multiple discounts application to single line. This is not the only place we do that and imo it should be corrected in separate PR